source/crud/crud.rst

```diff
@@ -661,9 +661,23 @@ older sharded clusters.
 estimatedDocumentCount
 ~~~~~~~~~~~~~~~~~~~~~~
 
-The estimatedDocumentCount function is implemented using the `count` command
-with no query filter, skip, limit, or other options that would alter the
-results. As documented above, the only supported option is maxTimeMS.
+On server versions greater than or equal to 4.9.0 (wire version 12 or higher),
```

Comment:
I tried this out against a non-existent collection, and got different results with aggregate and count:

```
MongoDB Enterprise > db.foo.count()
0
```

```
MongoDB Enterprise > pipeline = [
...     { '$collStats': { 'count': {} } },
...     { '$group': { '_id': 1, 'n': { '$sum': '$count' } } }
...   ]
[
	{
		"$collStats" : {
			"count" : {

			}
		}
	},
	{
		"$group" : {
			"_id" : 1,
			"n" : {
				"$sum" : "$count"
			}
		}
	}
]
MongoDB Enterprise > db.foo.aggregate(pipeline)
uncaught exception: Error: command failed: {
	"ok" : 0,
	"errmsg" : "PlanExecutor error during aggregation :: caused by :: Unable to retrieve count in $collStats stage: Collection [test.foo] not found.",
	"code" : 40481,
	"codeName" : "Location40481"
} : aggregate failed :
```

I think one way or another this behavioral difference needs to be hidden from applications.  Our options are:

1. Request a server change in the $collStats stage to not error for non-existent collections, at least when count is the only thing needed
2. Handle the specific error in drivers and swallow it, returning a count of 0.

If we choose the second option, we should request a stable error code with a label for this error.  Note that drivers already check for error code 26 in order to ignore failures to drop non-existent collections or indexes.

Reply:
@jyemin What server version were you testing on? My understanding is that [SEVER-35479](https://jira.mongodb.org/browse/SERVER-35479) was supposed to address this and return an empty result.

@benjirewis: Either way, I think we'd benefit from adding another test that uses a non-existent collection. You should be able to do that by creating a separate collection entity with a new namespace and _not_ specifying that namespace in `initialData`.

Reply:
I'm testing on 4.9.0-alpha-696-gb4de2e9

Reply:
Discussed offline. [SERVER-35479](https://jira.mongodb.org/browse/SERVER-35479) was resolved by making sure that a consistent `NamespaceNotFound` error was returned in the event of an aggregate with $collStats being run on a non-existent namespace. Since there appears to be precedent, drivers should swallow this specific error when running a `count` aggregate with $collStats, and return a count of 0.

Added two new spec tests to ensure a count of `0` with no error on 4.9+ and < 4.9. Added a line in the spec about swallowing the error.